esrally/track/params.py

```diff
@@ -763,7 +768,12 @@ def params(self):
         # self.internal_params always reads all files. This is necessary to ensure we terminate early in case
         # the user has specified ingest percentage.
         if self.current_bulk == self.total_bulks:
-            raise StopIteration()
+            # start from the beginning in looped mode, otherwise stop the run
+            if self.looped:
+                self.current_bulk = 0
+                self._init_internal_params()
```

Comment:
With looped mode, we will have the possibility to add an override for bulk and batch sizes in steps if it is something we can use in the future.

Reply:
The idea is looped operation should be limited at the task level, and you can have as many tasks with different batch sizes, etc. as you want. Looped mode basically means "keep on generating bulk requests until some external criteria stop the task".